src/figuresClasses.ts

```diff
@@ -1,19 +1,90 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  color: Color;
+
+  a: number;
 
-export class Circle {
+  b: number;
 
+  c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    if ((c >= a + b) || (a <= 0 || b <= 0 || c <= 0)) {
+      throw new Error();
+    }
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const square = (this.a + this.b + this.c) / 2;
+
+    return Math.floor(
+      Math.sqrt(
+        square * (square - this.a) * (square - this.b) * (square - this.c),
+      ) * 100,
+    ) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape;
+
+  color: Color;
 
+  radius: number;
+
+  constructor(color: Color, radius: number) {
+    if (radius <= 0) {
+      throw new Error();
+    }
+    this.shape = 'circle';
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    return Math.floor((Math.PI * (this.radius ** 2)) * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  width: number;
+
+  height: number;
+
+  constructor(color: Color, width: number, height: number) {
+    if (width <= 0 || height <= 0) {
+      throw new Error();
+    }
+    this.shape = 'rectangle';
+    this.color = color;
+    this.width = width;
+    this.height = height;
+  }
+
+  getArea(): number {
+    return Math.floor((this.width * this.height) * 100) / 100;
```

Comment:
it is better to remove logic of rounding to separate function & to use it everywhere